include/arch/arm/arch/model/smp.h

```diff
@@ -13,7 +13,11 @@
 #ifdef ENABLE_SMP_SUPPORT
 static inline cpu_id_t cpuIndexToID(word_t index)
 {
+#ifdef CONFIG_ARM_GIC_V3_SUPPORT
+    return index;
+#else
     return BIT(index);
+#endif
 }
```

Comment:
I'd prefer to remove this function altogether and do any `cpuIndexToID` operation required within `ipi_send_target` itself.

Reply:
Same, would it be OK to open a separate issue for this?

Reply:
I'm fine with a new PR that reworks `ipi_send_target` to only take a single target instead of a list, together with moving all `cpuIndexToID` into architecture specific code.

The only reason to do it the way you did is to hide any changes from verification. All it does is move the `cpuIndexToID` from the caller to where it's actually needed, I hope that's also a trivial change on the verification side.

Reply:
I've opened an issue [here](https://github.com/seL4/seL4/issues/1138) so that this does not get lost.